Core/clim-basic/clipboard.lisp

```diff
@@ -0,0 +1,174 @@
+PRESENTATION-TYPE. It is the responsibility of the backend to use the
+function CONVERT-CLIPBOARD-CONTENT to retrieve the underlying data
+from the object when needed. Note that the conversion operation may be
+expensive (in particular for images) so it is recommended that the a
+full conversion only be performed when a request to copy the content
+is actually performed. Use the :CHECK-ONLY argument to check whether a
+conversion is supported.
+
+PORT is the port that the sheet belongs to. It should be the primary
+argument that individual sheets specialise on.
+
+SHEET is the owner of the selection or clipboard.
+
+If CLIPBOARD-P is true, this call is a request to copy to the
+clipboard, if false it's a copy to the selection.
+
+OBJECT is the object that should be copied.
+
+PRESENTATION-TYPE is the CLIM presentation type for OBJECT. It should
+be passed to CONVERT-CLIPBOARD-CONTENT when the actual clipboard data
+is retrieved.")
+  (:method ((port clim:port) (sheet clim:sheet) object &key presentation-type)
+    (declare (ignore presentation-type))
+    nil))
+
+(defgeneric clear-selection (port sheet)
+  (:documentation "Clear the content of the selection.
+
+Please refer to the documentation for CLEAR-CLIPBOARD for details.")
+  (:method ((port clim:port) (sheet clim:sheet))
+    nil))
+
+(defgeneric clear-clipboard (port sheet)
+  (:documentation "Clear the content of the clipboard.
+
+When this function is called, the backend should ensure that the
+content of the selection or the clipboard is no longer available for
+pasting.
+
+PORT is the port that the sheet belongs to. It should be the primary
+argument that individual sheets specialise on.
+
+SHEET is the owner of the selection or clipboard.
+
+Clear the clipboard if CLIPBOARD-P is true, otherwise the selection.")
+  (:method ((port clim:port) (sheet clim:sheet))
+    nil))
+
+(defgeneric local-selection-content (port)
+  (:documentation "Returns the content of the selection in this Lisp image.
+If the global selection is currently held, the value returned is the
+same as what would be sent in response to a REQUEST-CLIPBOARD-CONTENT
+call.
+
+If content is available, this function should return two values: The
+content of the selection, and the presentation-type of the object.
+Otherwise NIL,NIL should be returned.
+
+This function is used by the copy selection to clipboard operation.
+This is in order to present the same beahviour as most non-CLIM
+applications exhibit: They are are able perform the copy
+operation (usually invoked by control-c) even when not owning the
+system-wide selection.
+
+The simplest implementation of this function would simply return the
+content of the selection, or NIL,NIL if it is not held.")
+  (:method ((port clim:port))
+    nil))
+
+;;;
+;;;  The following functions implement the standard API to request
+;;;  content from the clipboard.
+;;;
+
+(defgeneric request-clipboard-content (port pane type)
```

Comment:
generic function request-clipboard-content is defined twice (see below). the second docstring is of course more acceptable.

Reply:
Fixed.